docs/async.md

```diff
@@ -0,0 +1,210 @@
+# Async Client
```

Comment:
I think I like the title proposed in the navigation better… Matches `HTTP/2 Support` as well:

```suggestion
# Async Support
```

Reply:
Good call, yup!